File: 1/sol.py
#!/usr/bin/env python
from functools import reduce

"""
Aux function to reduce the lines
If the value is null, append 0 to the accumulator (new elf)
Otherwise, add the value to the last element of the list 
"""
def aux(acc, x):
    x = x.strip()
    if x == '':
        acc.append(0)
    else:
        acc[-1] += int(x)
    return acc

"""Return the calories carried by the strongest elves"""
def main():
    inp = open('input', 'r').readlines()
    calories = sorted(reduce(aux, inp, [0]), reverse=True)
    return calories[0], sum(calories[:3])

if __name__ == '__main__':
    top1, top3 = main()
    print(f"The strongest elf carries {top1} calories!")  # +1 because of python indexing
    print(f"The top 3 elves carry {top3} calories!")

File: 1/sol.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func check(e error) {
    if e != nil {
        panic(e)
    }
}

func max(l []uint64) (uint64, int) {
	var m uint64
	var im int
	for i, cal := range l {
	    if i==0 || cal > m {
		m = cal
		im = i
	    }
	}
	return m, im
}

func sort(l []uint64) []uint64 {
	sorted := l
	for k := range sorted {
		m, im := max(sorted[k:])
		sorted[im + k] = sorted[k]
		sorted[k] = m
	}
	return sorted
}

func sum(l []uint64) uint64 {
	s := uint64(0)
	for _, el := range l {
		s += el
	}
	return s
}

func main() {
	src, err := os.Open("input")
	defer src.Close()
	check(err)
	scanner := bufio.NewScanner(src)

	supplies := make([]uint64, 0)
	s := uint64(0)
	for scanner.Scan() {
		l := scanner.Text()
		if l == "" {
			supplies = append(supplies, s)
			s = 0
		} else {
			i, err := strconv.Atoi(l)
			check(err)
			s += uint64(i)
		}
	}
	supplies = append(supplies, s)
	supplies = sort(supplies)

	fmt.Printf("The strongest elf carries %d calories!\n", supplies[0])
	fmt.Printf("The top 3 elves carry %d calories!\n", sum(supplies[:3]))
}
